marketplace/interactions/serializers.py

```diff
@@ -18,3 +18,28 @@ def get_owned(self, obj) -> bool:
             return obj.created_by == request.user
 
         return False
+
+
+class RatingSerializer(AppTypeBaseSerializer):
+    class Meta:
+        model = Rating
+        fields = ["app_code", "uuid", "rate", "created_by", "modified_by"]
+        read_only_fields = ("app_code", "uuid")
+
+    def create(self, validated_data):
+        rate = validated_data.get("rate")
+
+        credentials = {
+            "created_by": validated_data.get("created_by"),
+            "app_code": validated_data.get("app_code"),
+            "defaults": {"rate": rate},
+        }
+
+        rating, created = Rating.objects.get_or_create(**credentials)
+
+        if not created:
+            rating.modified_by = validated_data.get("modified_by")
+            rating.rate = rate
+            rating.save()
```

Comment:
Use update_or_create instead: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#update-or-create

Reply:
looks like update_or_create is not working because of core.models.BaseModel.save(). Let's keep this way by now, but we are creating more and more complexity in things Django treats very well with no need to override.